frontend/src/components/widgets/CustomComponent/ComponentInstance.tsx

```diff
@@ -441,4 +443,17 @@ function tryGetValue(
   return obj.hasOwnProperty(name) ? obj[name] : defaultValue
 }
 
+// Exported for testing purposes.
+export function toComponentTheme(theme: Theme): any {
+  const { colors, genericFonts } = theme
+  return {
+    primaryColor: colors.primary,
+    secondaryColor: colors.secondary,
+    backgroundColor: colors.bgColor,
+    secondaryBackgroundColor: colors.secondaryBg,
+    textColor: colors.bodyText,
+    font: genericFonts.bodyFont,
```

Comment:
Totally not needed for this but I'm wondering if we should create a ThemeInput object as part of the Theme object so we don't have to keep converting this back and forth.

Reply:
:+1:  will note this down as a possible refactor for the near future

Reply:
I may or may not do it as part of the theme creator or at least some better way to handle this translation back and forth. It's ugly and gross.